### DISCUSSION TITLE
question about uv_cpu_info

### DISCUSSION THREAD

**Discussion by theanarkh** [libuv/libuv#4125] [Q&A]
Hi, all, what do the `cpu_times` field of `uv_cpu_info` function mean ? How to use this time to calculate the usage of each CPU core ?

**Comment by bnoordhuis** [comment]
The times are in milliseconds and the fields correspond to their counterparts documented in the /proc/stat section of [man 5 proc](https://man7.org/linux/man-pages/man5/proc.5.html).

**Reply by theanarkh** [reply]
Thanks. can i use this code to calculate the cpu usage ? 
```c
let lastTime = Date.now();
let lastUsage = uv_cpu_info();

setInterval(() => {
    const now = Date.now();
    const interval = now - lastTime;
    const usage = uv_cpu_info();
    const total = (usage.idle + usage.nice + usage.sys + usage.irq + usage.user) - (lastUsage.idle + lastUsage.nice + lastUsage.sys + lastUsage.irq + lastUsage.user);
    const usage = total / interval;
    lastTime = now;
    lastUsage = usage;
}, 10000);
```
